nemo/lightning/pytorch/callbacks/garbage_collection.py

```diff
@@ -0,0 +1,70 @@
+import gc
+from typing import Any
+
+import pytorch_lightning as pl
+from nemo.utils import logging
+
+
+class GarbageCollectionCallback(pl.Callback):
+    """Callback for synchronized manual Garbage Collection. This is required for distributed training
+    as all processes need to synchronize to garbage collect at the same time, without which
+    one process might hog or straggle all the rest of the processes.
+
+    Migration from NeMo 1.0:
+        When mitrating from NeMo1,
+            - gc_interval = 0 implied no GC, simply do not add this callback to the trainer
+            - gc_interval > 0, this config is maps => gc_interval_train
+
+            - env-var:NEMO_MANUAL_GC_IN_VALIDATION=0 or doesn't exist => Set gc_interval_val to a very high value that it does not practically run.
+            - env-var:NEMO_MANUAL_GC_IN_VALIDATION=1 => Set gc_interval_val to the same value as gc_interval
+
+        Moving from boolean flag (NEMO_MANUAL_GC_IN_VALIDATION) to integer is to allow user to set any value based on the size of the
```

Comment:
`user to set any value`
-> `user to set a specific interval`

Reply:
Good suggestion. 
Next rev.